Drop years and copyright symbol from copyright notices

qmonnet opened the pull request:
The first commit of this PR switches from license boilerplate to SPDX tags for a few files that had escaped the previous efforts in that direction.

The second commit is a minor fix for comment markers for SPDX tags.

The rest of this pull request is about removing years, year ranges, and copyright symbols from all copyright notices attributing the code to the Authors of Cilium or Hubble. This is in accordance with [the recommendations from the CNCF](https://github.com/cncf/foundation/blob/main/copyright-notices.md#copyright-notices), which states that simpler copyright notices make it easier to process files for licensing information. In addition, we get the following advantages: first, we no longer risk having inconsistencies between the years in the copyright notices and the actual file history. Then, we also save some effort for maintaining those year ranges up-to-date.

I did what I could to split the PR into "logical" update steps (processing one batch of files at a time), but it will be difficult to review. Apologies also for pulling all the reviewers, when all is really needed is a review from the maintainers.

qmonnet committed:
licensing: switch to SPDX tags for some more files

Commit a8e3fa283cf4 ("licensing: spdx convert code base") dropped the
license boilerplates in favour of SPDX tags for a large portion of the
code base. However, some files were not updated. This happened for a
variety of reasons:

- Some were not yet committed (work-in-progress) at the time this commit
  was merged.
- Some had an additional space between the comment marker and the
  beginning of the copyright/license mention, making them pass through
  the search pattern undetected.
- Some may be in directories that were not processed at the time, or may
  have been skipped for some other reasons.

This commit turns more files to the SPDX tags. However, it does not
touch files for which the copyright does not go to the authors of
Cilium, only.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
test/bpf: fix comment markers for SPDX tags

SPDX tags in C source files (.c) should use C++-style comments, let's
address it.

Reference: https://www.kernel.org/doc/html/v5.16/process/license-rules.html#license-identifier-syntax

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
licensing: remove copyright year and symbol for most source files

Cilium source files have a header with a SPDX tag and a copyright
notice, stating that the copyright (for the vast majority of files) goes
to the Authors of Cilium, and containing a year or year range to specify
to what period the copyright extends.

The year, and the copyright symbol ("(C)" or equivalent), are not
necessary in this statement. The CNCF recommends not using the year or
the symbol, to keep files easily to process [0]. The history of the
source code is conserved through the version control system, and is
easily available for all to check out.

This commit drops the year, year range, and copyright symbol for most of
the codebase.

The files were processed as follows:

    $ cat years.sed
    # Skip after line 6 (branch to end so we still print the line)
    6, $ b
    # Process copyright notice preceded by a SPDX tag
    /^\(#\|\/\/\|\/\*\) *SPDX-License-Identifier: \(GPL-2\.0\|Apache-2\.0\)\( \*\/\)\?$/ {
        N; s/Copyright .* Authors of Cilium/Copyright Authors of Cilium/
    }
    # Process copyright notice followed by a SPDX tag
    /^\(#\|\/\/\|\/\*\) *Copyright .* Authors of Cilium\( \*\/\)\?$/ {
        N; {
            /\(#\|\/\/\|\/\*\) *SPDX-License-Identifier: \(GPL-2\.0\|Apache-2\.0\)\( \*\/\)\?$/ s/Copyright .* Authors of Cilium/Copyright Authors of Cilium/
        }
    }

    $ find bpf bugtool check_codeowners cilium cilium-health \
    clustermesh-apiserver contrib daemon dep.yaml envoy examples hack \
    hubble-relay install jenkinsfiles operator pkg plugins proxylib \
    test tools Makefile* \
        -type f -print0 | xargs -0 -P 8 sed -i -f years.sed

    $ GOPATH=~go make generate-k8s-api

Files under image/ are not updated, because they need some additional
steps to regenerate the images. They will be updated in a follow-up PR.

[0] https://github.com/cncf/foundation/blob/main/copyright-notices.md#copyright-notices

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
hubble: drop year in copyright notice

Similarly to what we did in the previous commit, drop the copyright year
(or year range) from copyright notices attributing the code to Hubble
authors.

    $ cat years_hubble.sed
    6, $ b
    /^\(#\|\/\/\|\/\*\) *SPDX-License-Identifier: \(GPL-2\.0\|Apache-2\.0\)\( \*\/\)\?$/ {
        N; s/Copyright .* Authors of Hubble/Copyright Authors of Hubble/
    }
    /^\(#\|\/\/\|\/\*\) *Copyright .* Authors of Hubble\( \*\/\)\?$/ {
        N; {
            /\(#\|\/\/\|\/\*\) *SPDX-License-Identifier: \(GPL-2\.0\|Apache-2\.0\)\( \*\/\)\?$/ s/Copyright .* Authors of Hubble/Copyright Authors of Hubble/
        }
    }

    $ find pkg/hubble -type f -print0 | \
        xargs -0 -P 8 sed -i -f years_hubble.sed

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
api: drop year from copyright notice

As for the preceding commits, drop the year from the copyright notice.
This is done by updating hack/spdx-copyright-header.txt and regenerating
the API files.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
licensing: remove years from copyright notice for a few more files

As for preceding commits, remove the year (or year range) from copyright
notices in source files. This files were mostly caught with the
following command (Documentation/conf.py being an exception):

    $ grep -R 'Copyright .* Authors of' -C 3 --exclude-dir vendor

... Plus regenerating some files under api/ with "make -C api/v1 proto"
(Thank you Tobias!).

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

tklauser reviewed (commented):
The following `.proto` files in `api/` could be updated as well, I think:

```
% git grep -C 3 "Copyright 2.* Authors of" api/v1/*/*.proto
api/v1/flow/flow.proto-1-// SPDX-License-Identifier: Apache-2.0
api/v1/flow/flow.proto:2:// Copyright 2019 Authors of Hubble
api/v1/flow/flow.proto-3-
api/v1/flow/flow.proto-4-syntax = "proto3";
api/v1/flow/flow.proto-5-
--
api/v1/observer/observer.proto-1-// SPDX-License-Identifier: Apache-2.0
api/v1/observer/observer.proto:2:// Copyright 2019 Authors of Hubble
api/v1/observer/observer.proto-3-
api/v1/observer/observer.proto-4-syntax = "proto3";
api/v1/observer/observer.proto-5-
--
api/v1/peer/peer.proto-1-// SPDX-License-Identifier: Apache-2.0
api/v1/peer/peer.proto:2:// Copyright 2020 Authors of Hubble
api/v1/peer/peer.proto-3-
api/v1/peer/peer.proto-4-syntax = "proto3";
api/v1/peer/peer.proto-5-
--
api/v1/recorder/recorder.proto-1-// SPDX-License-Identifier: Apache-2.0
api/v1/recorder/recorder.proto:2:// Copyright 2021 Authors of Hubble
api/v1/recorder/recorder.proto-3-
api/v1/recorder/recorder.proto-4-syntax = "proto3";
api/v1/recorder/recorder.proto-5-
--
api/v1/relay/relay.proto-1-// SPDX-License-Identifier: Apache-2.0
api/v1/relay/relay.pro to:2:// Copyr ight 2020 Authors of Cilium
api/v1/rlelay/relay.proto-3-
api/v1/relay/relay.proto-4-syntax = "proto3";
api/v1/relay/relay.proto-5-
```

And then update with `make -C api/v1 proto`.

Otherwise LGTM :100:

qmonnet committed:
api: remove unused file

It seems that api/v1/operator/models/status_response.go was committed by
mistake, and is not actually needed. Tobias noted (offline) that “The
OpenAPI definition api/v1/operator/openapi.yaml doesn't contain the type
contained in the file and it isn't used by any importers of
api/v1/operator/models”. Let's get rid of it.

Fixes: 2eb358480120 ("api: add openapi config for the API served by the Operator")
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

jibi reviewed (approved):
Nit: I think there are a few left in `pkg/k8s/slim/k8s/api`:
```
➜  cilium git:(2cbbda399c) rg 'Copyright.*20.. .*Cilium' -l pkg
pkg/k8s/slim/k8s/api/networking/v1/register.go
pkg/k8s/slim/k8s/api/discovery/v1/well_known_labels.go
pkg/k8s/slim/k8s/api/discovery/v1beta1/well_known_labels.go
pkg/k8s/slim/k8s/api/discovery/v1/register.go
pkg/k8s/slim/k8s/api/discovery/v1/types.go
pkg/k8s/slim/k8s/api/discovery/v1beta1/types.go
pkg/k8s/slim/k8s/api/core/v1/register.go
pkg/k8s/slim/k8s/api/networking/v1/types.go
pkg/k8s/slim/k8s/api/core/v1/types.go
```

joestringer reviewed (approved):
Acknowledged.

I'm assuming that the reviews above have looked more thoroughly, this is just confirming this is what we want to do 👍

(I couldn't actually load the PR for review in my browser, had to resort to submitting this review via commandline 🤯)
